client/my-sites/term-tree-selector/terms.jsx

```diff
@@ -0,0 +1,406 @@
+
+		if ( this.props.termsHierarchy ) {
+			count += this.props.termsHierarchy.length;
+		}
+
+		if ( this.props.loading || ! this.props.terms ) {
+			count += 1;
+		}
+
+		return count;
+	},
+
+	onSearch( event ) {
+		const searchTerm = event.target.value;
+		if ( this.state.searchTerm && ! searchTerm ) {
+			this.props.onSearch( '' );
+		}
+
+		if ( searchTerm === this.state.searchTerm ) {
+			return;
+		}
+
+		if ( ! this.hasPerformedSearch ) {
+			this.hasPerformedSearch = true;
+			analytics.ga.recordEvent( this.props.analyticsPrefix, 'Performed Term Search' );
+		}
+
+		this.setState( { searchTerm } );
+		this.debouncedSearch();
+	},
+
+	setVirtualScrollRef( ref ) {
+		this.virtualScroll = ref;
+	},
+
+	renderItem( item ) {
+		const onChange = ( ...args ) => this.props.onChange( item, ...args );
+		const setItemRef = ( ...args ) => this.setItemRef( item, ...args );
+
+		const { multiple, defaultTermId, translate, selected } = this.props;
+		const itemId = item.ID;
+		const name = decodeEntities( item.name ) || translate( 'Untitled' );
+		const checked = includes( selected, itemId );
+		const inputType = multiple ? 'checkbox' : 'radio';
+		const disabled = (
+			multiple &&
+			checked &&
+			defaultTermId &&
+			1 === selected.length &&
+			defaultTermId === itemId
+		);
+
+		const input = (
+			<input
+				type={ inputType }
+				value={ itemId }
+				onChange={ onChange }
+				disabled={ disabled }
+				checked={ checked }
+			/>
+		);
+
+		return (
+			<div
+				key={ itemId }
+				ref={ setItemRef }
+				className="term-tree-selector__list-item">
+				<label>
+					{ input }
+					<span className="term-tree-selector__label">{ name }</span>
+				</label>
+				{ item.items && (
+					<div className="term-tree-selector__nested-list">
+						{ item.items.map( this.renderItem ) }
+					</div>
+				) }
+			</div>
+		);
+	},
+
+	renderNoResults() {
+		if ( this.hasNoSearchResults() || this.hasNoTerms() ) {
+			return (
+				<div key="no-results" className="term-tree-selector__list-item is-empty">
+					{ ( this.hasNoSearchResults() || ! this.props.emptyMessage ) && (
+						<NoResults
+							createLink={ this.props.createLink } />
+					) }
+					{ this.hasNoTerms() && this.props.emptyMessage }
+				</div>
+			);
+		}
+	},
+
+	renderRow( { index } ) {
+		const item = this.getItem( index );
+		if ( item ) {
+			return this.renderItem( item );
+		}
+
+		return (
+			<div key="placeholder" className="term-tree-selector__list-item is-placeholder">
+				<label>
+					<input
+						type={ this.props.multiple ? 'checkbox' : 'radio' }
+						disabled
+						className="term-tree-selector__input" />
+					<span className="term-tree-selector__label">
+						{ this.props.translate( 'Loading…' ) }
+					</span>
+				</label>
+			</div>
+		);
+	},
+
+	render() {
+		const rowCount = this.getRowCount();
+		const isCompact = this.isCompact();
+		const searchLength = this.state.searchTerm.length;
+		const showSearch = ( searchLength > 0 || ! isCompact ) &&
```

Comment:
The additional condition is to prevent the search box from being shown on the initial load - just like it currently acts.